wp_api/src/wp_com/support_tickets.rs

```diff
@@ -0,0 +1,145 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::{date::WpGmtDateTime, impl_as_query_value_for_new_type};
+
+#[derive(Debug, PartialEq, Eq, Serialize, uniffi::Record)]
+pub struct CreateSupportTicketParams {
+    pub subject: String,
+    pub message: String,
+    pub application: String,
+    #[uniffi(default = None)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub wpcom_site_id: Option<u64>,
+    #[uniffi(default = [])]
+    pub tags: Vec<String>,
+    #[uniffi(default = [])]
+    pub attachments: Vec<String>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, uniffi::Record)]
+pub struct SupportConversationSummary {
+    pub id: ConversationId,
+    pub title: String,
+    pub description: String,
+    pub status: String,
+    pub created_at: WpGmtDateTime,
+    pub updated_at: WpGmtDateTime,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, uniffi::Record)]
+pub struct SupportConversation {
+    pub id: ConversationId,
+    pub title: String,
+    pub description: String,
+    pub status: String,
+    pub created_at: WpGmtDateTime,
+    pub updated_at: WpGmtDateTime,
+    pub messages: Vec<SupportMessage>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, uniffi::Record)]
+pub struct SupportMessage {
+    pub id: u64,
+    pub content: String,
+    pub author: SupportMessageAuthor,
+    pub role: String,
```

Comment:
At the moment how the `author` is parsed id depending on the `SupportMessageAuthor` enum variant order: it's a "User" if the JSON can be parsed as an "User"; then it's a "Agent" if the JSON can be parsed as an "Agent". Would it be more correct to parse `author` based on the `role` value, instead of how the `SupportMessageAuthor` enum type is declared?

Reply:
> or separating the SupportMessage type as SupportUserMessage and SupportAgentMessage. 

This sounds good to me 👍

Reply:
I have created https://github.com/Automattic/wordpress-rs/issues/852 to track the potential issue discussed above.